doc/ecosystem.md

```diff
@@ -0,0 +1,132 @@
+
+PyGMT relies on IPython to provide a rich interactive experience in Jupyter notebooks.
+
+#### GeoPandas
+
+[GeoPandas][] is an open source project to make working with geospatial data in Python
+easier. GeoPandas extends the datatypes used by [pandas][] to allow spatial operations
+on geometric types. Geometric operations are performed by [Shapely][]. GeoPandas further
+depends on [pyogrio][] for file access and [Matplotlib][] for plotting.
+
+PyGMT doesn't directly rely on GeoPandas, but provides support of GeoPandas's two main
+data structure, {class}`geopandas.GeoDataFrame` and {class}`geopandas.GeoSeries`, which
+can be directly used in data processing and plotting functions/methods of PyGMT.
+
+#### contextily
+
+[contextily][] is a small Python package to retrieve tile maps from the internet. It can
+add those tiles as basemap to matplotlib figures or write tile maps to disk into
+geospatial raster files. Bounding boxes can be passed in both WGS84 (EPSG:4326) and
+Spheric Mercator (EPSG:3857).
+
+In PyGMT, {func}`pygmt.datasets.load_tile_map` and {class}`pygmt.Figure.tilemap` rely
+on it.
+
+#### rioxarray
+
+[rioxarray][] is a geospatial [Xarray][] extension powered by [rasterio][]. Built on top
+of rasterio, it enables seamless reading, writing, and manipulation of multi-dimensional
+arrays with geospatial attributes such as coordinate reference systems (CRS) and spatial
+extent (bounds).
+
+Currently, PyGMT relies on [rioxarray][] to save multi-band rasters to temporary files
+in GeoTIFF format, to support processing and plotting 3-D {class}`xarray.DataArray`
+images.
+
+```{note}
+We're working towards removing the dependency of the [rioxarray][] package in
+[PR #3468](https://github.com/GenericMappingTools/pygmt/pull/3468).
+```
+
+#### PyArrow
+
+[Apache Arrow][] is a development platform for in-memory analytics. It contains a set of
+technologies that enable big data systems to process and move data fast. It specifies a
+standardized language-independent columnar memory format for flat and hierarchical data,
+organized for efficient analytic operations on modern hardware. The Arrow Python bindings
+(also named "PyArrow") have first-class integration with NumPy, pandas, and built-in
+Python objects. They are based on the C++ implementation of Arrow.
+
+```{note}
+If you have [PyArrow][] installed, PyGMT does have some initial support for
+`pandas.Series` and `pandas.DataFrame` objects with Apache Arrow-backed arrays.
+Specifically, only uint/int/float, date32/date64 and string types are supported for now.
+Support for Duration types and GeoArrow geometry types is still a work in progress.
+For more details, see
+[issue #2800](https://github.com/GenericMappingTools/pygmt/issues/2800).
+```
+
+## PyGMT ecosystem
+
+Various packages rely on PyGMT for geospatial data processing, analysis, and visualization.
+Below is an incomplete list (in no particular order) of tools which form the PyGMT-related
+ecosystem.
```

Comment:
For a package that rely on PyGMT, I think a short description about the package itself and how PyGMT is used in the package is enough. 

For example, I know [SHTOOLS](https://github.com/SHTOOLS/SHTOOLS) maintained by `MarkWieczorek` uses PyGMT. I expect to see it listed here like below:
```
### SHTOOLS

pyshtools is an archive of Python software that can be used for spherical harmonic transforms, multitaper spectral analyses, expansions of gridded data into Slepian basis functions, and standard operations on global gravitational and magnetic field data.

It uses PyGMT for the creation of publication quality maps.
```

I don't plan to add SHTOOLS myself. Instead, after merging this PR, I prefer to ping `MarkWieczorek` to see if he is willing to add SHTOOLS himself.

Reply:
Sounds good to me.